python/cudf/cudf/pandas/__init__.py

```diff
@@ -50,17 +52,30 @@ def install():
             mr = rmm.mr.CudaAsyncMemoryResource(initial_pool_size=free_memory)
             rmm.mr.set_current_device_resource(mr)
         elif rmm_mode == "managed":
-            mr = rmm.mr.ManagedMemoryResource()
+            mr = rmm.mr.PrefetchResourceAdaptor(rmm.mr.ManagedMemoryResource())
             rmm.mr.set_current_device_resource(mr)
         elif rmm_mode == "managed_pool":
-            mr = rmm.mr.PoolMemoryResource(
-                rmm.mr.ManagedMemoryResource(),
-                initial_pool_size=free_memory,
+            mr = rmm.mr.PrefetchResourceAdaptor(
+                rmm.mr.PoolMemoryResource(
+                    rmm.mr.ManagedMemoryResource(),
+                    initial_pool_size=free_memory,
+                )
             )
             rmm.mr.set_current_device_resource(mr)
```

Comment:
Can we move the `set_current_device_resource` calls outside the conditional? The only conditional part is the actual construction of `mr`, so if we default `mr` to `current_mr` we can unconditionally call `set_current_device_resource` after the conditional.

Reply:
Done